### ISSUE TITLE
Crash with 1.10.2 (Latest Barrel Version)

### ISSUE THREAD

**Issue by Speiger** [shadowfacts/YeOldeTanks#27]
```
java.lang.IllegalArgumentException: No TileEntity at position BlockPos{x=251, y=66, z=240}
	at net.shadowfacts.shadowmc.block.TileEntityProvider.getTileEntity(TileEntityProvider.java:32)
	at net.shadowfacts.yeoldetanks.block.base.BlockBarrelBase.getLightValue(BlockBarrelBase.java:147)
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.getLightValue(BlockStateContainer.java:481)
	at net.minecraft.block.Block.func_185484_c(Block.java:382)
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185889_a(BlockStateContainer.java:349)
	at team.chisel.client.ChiselExtendedState.func_185889_a(ChiselExtendedState.java:41)
	at net.minecraftforge.client.model.pipeline.VertexLighterFlat.updateLightmap(VertexLighterFlat.java:244)
	at net.minecraftforge.client.model.pipeline.VertexLighterFlat.processQuad(VertexLighterFlat.java:165)
	at net.minecraftforge.client.model.pipeline.QuadGatheringTransformer.put(QuadGatheringTransformer.java:63)
	at net.minecraftforge.client.model.pipeline.LightUtil.putBakedQuad(LightUtil.java:133)
	at net.minecraft.client.renderer.block.model.BakedQuad.pipe(BakedQuad.java:66)
	at net.minecraftforge.client.model.pipeline.ForgeBlockModelRenderer.render(ForgeBlockModelRenderer.java:124)
	at net.minecraftforge.client.model.pipeline.ForgeBlockModelRenderer.func_187497_c(ForgeBlockModelRenderer.java:82)
	at net.minecraft.client.renderer.BlockModelRenderer.func_187493_a(BlockModelRenderer.java:46)
	at net.minecraft.client.renderer.BlockModelRenderer.func_178267_a(BlockModelRenderer.java:37)
	at net.minecraft.client.renderer.BlockRendererDispatcher.func_175018_a(BlockRendererDispatcher.java:81)
	at gcewing.architecture.BaseAORenderingManager$CustomBlockRendererDispatcher.func_175018_a(BaseAORenderingManager.java:85)
	at net.minecraft.client.renderer.chunk.RenderChunk.func_178581_b(RenderChunk.java:199)
	at net.minecraft.client.renderer.chunk.ChunkRenderWorker.func_178474_a(SourceFile:100)
	at net.minecraft.client.renderer.chunk.ChunkRenderDispatcher.func_178505_b(ChunkRenderDispatcher.java:171)
	at net.minecraft.client.renderer.RenderGlobal.func_174970_a(RenderGlobal.java:952)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1286)
	at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1229)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1038)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1076)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
```
I just placed a barrel down it instantly Crashes and the World is basically no longer playeable.

**Comment by shadowfacts** [comment]
I'm unable to reproduce this either with just YOT (and dependencies) or with Chisel and ArchitectureCraft as well.

I don't believe the cause of this can be YOT, because, somehow, the perfectly valid assumption that whenever the block exists in the world, the TileEntity also exists is false. You should report this to the ArchitectureCraft dev and/or the Chisel devs.

**Comment by gcewing** [comment]
Were there any ArchitectureCraft blocks near where you placed the barrel?

If not, I'm doubtful whether this has anything to do with AC. The fact that AC's rendering dispatcher appears in the traceback means very little, since it's involved in any rendering operation when AC is installed. Non-AC blocks are just passed on to the base rendering code.

If someone can provide a recipe for reproducing this, I'll look into it further, otherwise there's not much I can do.

**Comment by Speiger** [comment]
@gcewing @shadowfacts i created a new world. Even a flat world and placed down a barrel = RenderCrash....
Nothing with other mods to interact with. This means its a issue with Overrides on top of each other
And no Sanitiy Checks are added.

**Comment by Speiger** [comment]
Since @shadowfacts your Crash is at the first place you should check your code where its crashing and not push it to the next form. Meanwhile @gcewing look into the code (your line) what you do there and provide information.
I am a coder too but i am not here to fix your guys code.

**Comment by gcewing** [comment]
Speiger wrote:
> i created a new world. Even a flat 
> world and placed down a barrel = RenderCrash....

Was this with or without AC? If with AC, does it still happen if AC is removed?

-- 
Greg

**Comment by gcewing** [comment]
Speiger wrote:
> Meanwhile @gcewing <https://github.com/gcewing> look into 
> the code (your line) what you do there and provide information.

This is what I'm doing there:

   public boolean renderBlock(IBlockState state, BlockPos pos, IBlockAccess
     world, VertexBuffer tess)
   {
     ICustomRenderer rend = getCustomRenderer(world, pos, state);
     if (rend != null)
       return customRenderBlockToWorld(world, pos, state, tess, null, rend);
     else
       return base.renderBlock(state, pos, world, tess); // <-- Here
   }

All I'm doing is passing what was given to me on to the vanilla
rendering code.

-- 
Greg

**Comment by gcewing** [comment]
I can't reproduce this either, with or without ArchitectureCraft.

Spieger, can you post the entire crash log, so we can see exactly what versions of everything is being used?

**Comment by Speiger** [comment]
http://imgur.com/a/5GvgI
My pack. Every mod i use. Right now YeOldBarrels is not installed because it crashes with it.
But i think you need to add a sanity check.

**Comment by Oeniarku** [comment]
I think the way to reproduce the crash is to disable the Smooth Lighting. Tested it on clean 1.10.2, forge 12.18.3.2281 and YeOldeTanks-1.10.2-2.0.1 + ShadowMC-1.10.2-3.6.1. Getting a very similar error.

**Comment by Speiger** [comment]
Yeah smooth lighting is disabled for me because my CPU and GPU are shit...
So you dont handle that case.

**Comment by shadowfacts** [comment]
Forgot to close this, but it was already fixed by 70cfbe45e9e24f93fae48dd9f952cb1221dd4f80.
